### PR TITLE
[#175844896] io-pay-portal-fe skeleton

### PR DESCRIPTION
**Pull request by infantesimone**
### Description

The goal of this PR is to provide a simple frontend that interacts with the services implemented in _io-functions-pay-portal_. 

### Proposed changes

1. Simple form that given a _Codice Avviso Pagamento_  validates the payment invoking the API:

   GET /payment-requests/{rtdId}

   implemented in _io-functions-pay-portal_. 

 image

 image

2. The CI/CD pipelines are defined in the _.devops_ folder. It is required to set the following variables on Azure DevOps:

- GIT_EMAIL
- GIT_USERNAME
- GITHUB_CONNECTION
- IO_PAY_PORTAL_FUNCTION
- PRODUCTION_AZURE_SUBSCRIPTION
- STAGING_AZURE_SUBSCRIPTION
- PRODUCTION_RESOURCE_GROUP_NAME
- PRODUCTION_CDN_ENDPOINT
- PRODUCTION_CDN_PROFILE_NAME 

### Usage

1. Clone the repo
   ```sh
   git clone https://github.com/pagopa/io-pay-portal
   ```
2. Install node packages
   ```sh
   yarn install
   ```
3. Export env variables from .env.io-pay-portal.development
   ```JS
    export $(grep -v '^#' .env.io-pay-portal.development | xargs) && chmod +x env.sh && source env.sh
   ```
4. Generate api client 
   ```sh
   yarn generate
   ```
5. Build 
   ```sh
   yarn build
6. tests 
   ```sh
   yarn test
   ```
7. Linter 
   ```sh
   yarn lint
8. Local dev server
   ```sh
   yarn dev
Then, point your browser to

- http://localhost:1234 

to reach the frontend.

### CODE REVIEW

File: io-pay-portal-fe/src/api/__tests__/services.test.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+describe("withApiRequestWrapper", () => {
+  it("should return a ", async () => {
```
Comment (GiovanniMancini) [inline io-pay-portal-fe/src/api/__tests__/services.test.ts:2]: Obviously this is just a stub test to develop the CI/CD pipelines. Can you confirm?

**Review by GiovanniMancini — APPROVED** [review]

File: io-pay-portal-fe/src/api/services.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+import { IResponseType } from "italia-ts-commons/lib/requests";
+import { PaymentRequestsGetResponse } from "../../generated/PaymentRequestsGetResponse";
+import { apiClient } from "./client";
+import {
+  ErrorResponses,
+  toDefaultResponseErrorInternal,
+  toErrorServerResponse,
+} from "./responses";
+
+export const withApiRequestWrapper = <T>(
```
Comment (cloudify) [inline io-pay-portal-fe/src/api/services.ts:17]: I'm not sure we actually need this logic in the frontend

**Review by cloudify — COMMENTED** [review]
@infantesimone sorry I've been late to the review, can you check my commments?

File: io-pay-portal-fe/src/api/services.ts (hunk 2 above)
Comment (infantesimone) [reply io-pay-portal-fe/src/api/services.ts:17]: `withApiRequestWrapper` is a method to check the status code of the api, actually I can simplify the returned `ErrorsResponse` 👍

**Comment by infantesimone** [comment]
I create a new PR for the latest comments @cloudify

**Pull request merged**